src/NuGet.Core/NuGet.Commands/SignCommand/CertificateProvider.cs

```diff
@@ -29,15 +39,42 @@ public static X509Certificate2Collection GetCertificates(CertificateSourceOption
             {
                 try
                 {
-                    var cert = new X509Certificate2(options.CertificatePath, options.CertificatePassword);
-                    return new X509Certificate2Collection(cert);
+                    if (!string.IsNullOrEmpty(options.CertificatePassword))
```

Comment:
You can simplify this by having 1 control flow:

    X509Certificate2 cert;

    if (...) { } else { }

    return new X509Certificate2Collection(cert);